### ISSUE TITLE
v1.10.0-RC1: Wifi disconnection causes reboot

### ISSUE THREAD

**Issue by ojousima** [ruuvi/ruuvi.gateway_esp.c#465]
This seems really random, occurs a few times a day according to uptime statistics. While it is not a problem for most users, sometimes WiFi fails to reconnect at boot at least on v1.9.2.

```
[0m
[0;32mI (63074803) esp-x509-crt-bundle: Certificate validated[0m
[0;32mI (63075982) http: [adv_post_task] HTTP POST to URL=https://network.ruuvi.com/record: STATUS=200[0m
[0;32mI (63075987) LEDS: [adv_post_task] leds_indication_on_network_ok[0m
[0;32mI (63075988) LEDS: [adv_post_task] LED: OFF[0m
[0;32mI (63082852) ruuvi_gateway: [main] free heap: 121248[0m
[0;32mI (63084406) ADV_POST_TASK: [adv_post_task] No networking for 3600 seconds - reboot the gateway[0m
I (63084407) wifi:state: run -> init (0)
I (63084409) wifi:pm stop, total sleep time: 0 us / 63080133480 us

W (63084415) wifi:<ba-del>idx
W (63084417) wifi:<ba-del>idx
W (63084420) wifi:<ba-del>idx
W (63084423) wifi:<ba-del>idx
I (63084426) wifi:new:<11,0>, old:<11,0>, ap:<255,255>, sta:<11,0>, prof:1
W (63084434) wifi:hmac tx: ifx0 stop, discard
W (63084437) wifi:hmac tx: ifx0 stop, discard
[0;32mI (63084441) wifi_manager: [sys_evt] WIFI_EVENT_STA_DISCONNECTED[0m
[0;32mI (63084447) wifi_manager: [sys_evt] WIFI_EVENT_STA_STOP[0m
[0;32mI (63084453) wifi_manager: [wifi_manager] MESSAGE: EVENT_STA_DISCONNECTED with Reason code: 8[0m
[0;31mE (63084454) TRANS_TCP: tcp_poll_read select error 113, errno = Software caused connection abort, fd = 54[0m
[0;32mI (63084463) wifi_manager: [wifi_manager] lost connection[0m
[0;31mE (63084473) MQTT_CLIENT: Poll read error: 119, aborting connection[0m
[0;32mI (63084479) wifi_manager: [wifi_manager] Set STA IP String to: 0.0.0.0[0m
[0;32mI (63084487) MQTT: [mqtt_task] MQTT_EVENT_DISCONNECTED[0m
[0;33mW (63084494) ruuvi_gateway: [wifi_manager] Wifi disconnected[0m
[0;32mI (63084499) LEDS: [mqtt_task] leds_indication_network_no_connection[0m
[0;32mI (63084506) LEDS: [wifi_manager] leds_indication_network_no_connection[0m
[0;32mI (63084513) LEDS: [mqtt_task] LED: Start blinking, interval: 200 ms, duty cycle: 50%[0m
[0;32mI (63084520) LEDS: [wifi_manager] LED: Start blinking, interval: 200 ms, duty cycle: 50%[0m
[0;32mI (63084537) wifi_manager: [wifi_manager] MESSAGE: ORDER_CONNECT_STA[0m
[0;31mE (63084544) wifi_manager: [wifi_manager] ../components/esp32-wifi-manager/src/wifi_manager_handle_msg.c:130 {wifi_handle_cmd_connect_sta}: esp_wifi_set_config failed, err=12308 (ESP_ERR_WIFI_STOP_STATE)[0m
I (63084563) wifi:flush txq
I (63084565) wifi:stop sw txq
I (63084568) wifi:lmac stop hw txq
[0;32mI (63ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x17 (SPI_FAST_FLASH_BOOT)
```

**Comment by ojousima** [comment]
Is this something that we can fix quickly for v1.10 or should we add this to v1.11?

**Comment by TheSomeMan** [comment]
> Is this something that we can fix quickly for v1.10 or should we add this to v1.11?

I can't explain how this could happen. 
After successful HTTP POST the timestamp is updated, the network watchdog timeout is calculated relative to this timestamp every second. 
In this log we can see that the last HTTP POST was performed at 63075982:
```
(63075982) http: [adv_post_task] HTTP POST to URL=https://network.ruuvi.com/record: STATUS=200
```

then, after 8424 ms the network watchdog was triggered:
```
(63084406) ADV_POST_TASK: [adv_post_task] No networking for 3600 seconds - reboot the gateway
```

**Comment by TheSomeMan** [comment]
Related issue #467 

I've found one problem - race condition between "adv_post" and "mqtt" threads can cause such reboots (but only if MQTT is enabled).

**Comment by ojousima** [comment]
The Gateway had MQTT + HTTP enabled, I can try to leave a gateway with only HTTP enabled to see if the problem remains

**Comment by TheSomeMan** [comment]
Fixed in #467
